packages/code-studio/src/settings/ColumnSpecificSectionContent.tsx

```diff
@@ -0,0 +1,818 @@
+  defaultDecimalFormatOptions: FormatOption;
+  defaultIntegerFormatOptions: FormatOption;
+  truncateNumbersWithPound: boolean;
+  timestampAtMenuOpen: Date;
+}
+
+export class ColumnSpecificSectionContent extends PureComponent<
+  ColumnSpecificSectionContentProps,
+  ColumnSpecificSectionContentState
+> {
+  static defaultProps = {
+    scrollTo: (): void => undefined,
+    defaults: {
+      defaultDateTimeFormat:
+        DateTimeColumnFormatter.DEFAULT_DATETIME_FORMAT_STRING,
+      defaultDecimalFormatOptions: {
+        defaultFormatString: DecimalColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      defaultIntegerFormatOptions: {
+        defaultFormatString: IntegerColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      showTimeZone: false,
+      showTSeparator: true,
+      timeZone: DateTimeColumnFormatter.DEFAULT_TIME_ZONE_ID,
+    },
+  };
+
+  static inputDebounceTime = 250;
+
+  static focusFirstInputInContainer(container: HTMLElement | null): void {
+    const input: HTMLElement | null | undefined = container?.querySelector(
+      'input, select, textarea'
+    );
+    if (input) {
+      input.focus();
+    }
+  }
+
+  static isSameOptions(
+    options1: FormatOption,
+    options2: FormatOption
+  ): boolean {
+    return options1.defaultFormatString === options2.defaultFormatString;
+  }
+
+  static isSameDecimalOptions(
+    options1: FormatOption,
+    options2: FormatOption
+  ): boolean {
+    return ColumnSpecificSectionContent.isSameOptions(options1, options2);
+  }
+
+  static isSameIntegerOptions(
+    options1: FormatOption,
+    options2: FormatOption
+  ): boolean {
+    return ColumnSpecificSectionContent.isSameOptions(options1, options2);
+  }
+
+  static isValidColumnName(name: string): boolean {
+    return name !== '' && DbNameValidator.isValidColumnName(name);
+  }
+
+  static isValidFormat(
+    columnType: string,
+    format: Partial<TableColumnFormat>
+  ): boolean {
+    // Undefined or empty string formats are always invalid
+    if (
+      !columnType ||
+      format.formatString == null ||
+      !isFormatStringFormat(format)
+    ) {
+      return false;
+    }
+    switch (columnType) {
+      case 'datetime':
+        return DateTimeColumnFormatter.isValid(format);
+      case 'decimal':
+        return DecimalColumnFormatter.isValid(format);
+      case 'int':
+        return IntegerColumnFormatter.isValid(format);
+      default: {
+        log.warn('Trying to validate format of unknown type');
+        return true;
+      }
+    }
+  }
+
+  static removeFormatRuleExtraProps(
+    item: FormatterItem
+  ): Omit<FormatterItem, 'id' | 'isNewRule'> {
+    const { id, isNewRule, ...rest } = item;
+    return rest;
+  }
+
+  static isFormatRuleValidForSave(rule: FormatterItem): boolean {
+    return (
+      ColumnSpecificSectionContent.isValidColumnName(rule.columnName) &&
+      ColumnSpecificSectionContent.isValidFormat(rule.columnType, rule.format)
+    );
+  }
```

Comment:
I think all of these static methods are also used in `FormattingSectionContent`. You should refactor these into their own utils file, e.g. `SettingsUtils.ts`. See `FormatterUtils.ts` for an example of a utils file (exporting a bunch of functions).